src/meta/processors/usersMan/AuthenticationProcessor.cpp

```diff
@@ -0,0 +1,271 @@
+    data.emplace_back(std::move(userKey),
+                      MetaServiceUtils::replaceUserVal(user, val));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    resp_.set_id(to(userId, EntryType::USER));
+    doPut(std::move(data));
+}
+
+
+void DropUserProcessor::process(const cpp2::DropUserReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    auto ret = getUserId(req.get_account());
+    if (!ret.ok()) {
+        if (req.get_missing_ok()) {
+            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+        } else {
+            LOG(ERROR) << "Drop User Failed :" << req.get_account() << " not found.";
+            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        }
+        onFinished();
+        return;
+    }
+    std::vector<std::string> keys;
+    keys.emplace_back(MetaServiceUtils::indexUserKey(req.get_account()));
+    keys.emplace_back(MetaServiceUtils::userKey(ret.value()));
+
+    // Collect related roles by user.
+    std::unique_ptr<kvstore::KVIterator> iter;
+    std::string prefix = "__roles__";
+    auto userRet = kvstore_->prefix(kDefaultSpaceId_, kDefaultPartId_, prefix, &iter);
+    if (userRet == kvstore::ResultCode::SUCCEEDED) {
+        while (iter->valid()) {
+            auto key = iter->key();
+            auto userId = MetaServiceUtils::parseRoleUserId(key);
+            if (userId == ret.value()) {
+                keys.emplace_back(std::move(key));
+            }
+            iter->next();
+        }
+    }
+
+    resp_.set_id(to(ret.value(), EntryType::USER));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    LOG(INFO) << "Drop User " << req.get_account();
+    doMultiRemove(std::move(keys));
+}
+
+
+void GrantProcessor::process(const cpp2::GrantRoleReq& req) {
+    const auto& roleItem = req.get_role_item();
+    CHECK_SPACE_ID_AND_RETURN(roleItem.get_space_id());
+    CHECK_USER_ID_AND_RETURN(roleItem.get_user_id());
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    std::vector<kvstore::KV> data;
+    data.emplace_back(MetaServiceUtils::roleKey(roleItem.get_space_id(), roleItem.get_user_id()),
+                      MetaServiceUtils::roleVal(roleItem.get_role_type()));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    doPut(std::move(data));
+}
+
+
+void RevokeProcessor::process(const cpp2::RevokeRoleReq& req) {
+    const auto& roleItem = req.get_role_item();
+    CHECK_SPACE_ID_AND_RETURN(roleItem.get_space_id());
+    CHECK_USER_ID_AND_RETURN(roleItem.get_user_id());
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
+    auto roleKey = MetaServiceUtils::roleKey(roleItem.get_space_id(), roleItem.get_user_id());
+    resp_.set_id(to(roleItem.get_user_id(), EntryType::USER));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    doRemove(std::move(roleKey));
+}
+
+
+void ChangePasswordProcessor::process(const cpp2::ChangePasswordReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::userLock());
```

Comment:
Change pwd is a special sentence?

Reply:
Yes, Change password is a separate sentence. No need to verify the password if it is a `GOD` role. else the password must be validated. And `GOD` can change anyone's password. Other roles can only change their own password.

```
change_password_sentence
    : CHANGE PASSWORD account TO new_password
    | CHANGE PASSWORD account FROM old_password TO new_password
    ;
```